src/java/org/apache/cassandra/cql3/functions/types/GettableByIndexData.java

```diff
@@ -465,6 +465,21 @@ public interface GettableByIndexData
      */
     public <K, V> Map<K, V> getMap(int i, TypeToken<K> keysType, TypeToken<V> valuesType);
 
+    /**
+     * Returns the {@code i}th value as a vector.
+     *
+     * <p>This method uses the {@link CodecRegistry} to find a codec to convert the underlying CQL
+     * type to a {@link List} (if the CQL type is a UDT, the registry will generate a codec
```

Comment:
You mention UDT here?

Reply:
I've wrongly copied that comment from the getter for UDTs. Revisiting this has made me realise that the getters/setters are incorrect for usage with nested types. I have just fixed them and added a new test case covering them.

Reply:
Nice catch! I hadn't gone that deep yet. Glad you caught it :-)